pkg/update/update_cockroach_version_common.go

```diff
@@ -46,39 +48,66 @@ func makeUpdateCockroachVersionFunction(
 	}
 }
 
+// TODO some of these should probably be panics or cancel the update at least
+// If we cannot find the Pod, or if we cannot find the container.
+// We need to return a status and an error instead of just an error.
+
 // makeWaitUntilCRDBPodIsRunningNewVersionFunction takes a cockroachImage and
 // returns a function which takes a Kubernetes clientset, a statefulset, and a
 // pod number within the statefulset. This function checks if the specified
 // pod is running the new cockroachImage version and is in a `ready` state.
 func makeIsCRBPodIsRunningNewVersionFunction(
 	cockroachImage string,
-) func(update *UpdateSts, podNumber int, l *zap.Logger) error {
-	return func(update *UpdateSts, podNumber int, l *zap.Logger) error {
+) func(update *UpdateSts, podNumber int, l logr.Logger) error {
+	return func(update *UpdateSts, podNumber int, l logr.Logger) error {
 		sts := update.sts
 		stsName := sts.Name
 		stsNamespace := sts.Namespace
 		podName := fmt.Sprintf("%s-%d", stsName, podNumber)
 		clientset := update.clientset
-		crdbPod, err := clientset.CoreV1().Pods(stsNamespace).Get(context.TODO(), stsName, metav1.GetOptions{})
-		if err != nil {
-			return errors.Wrapf(err, "error getting cockroach pod")
+
+		crdbPod, err := clientset.CoreV1().Pods(stsNamespace).Get(update.ctx, podName, metav1.GetOptions{})
+		if k8sErrors.IsNotFound(err) { // this is not an error
```

Comment:
This looks like the same code as `handleStsError`?

Reply:
Why is not found not an error but handled as an error?

Reply:
We talked on slack about this